examples/src/main/java/com/mapbox/navigation/examples/core/MultipleStopsActivity.kt

```diff
@@ -161,6 +165,19 @@ class MultipleStopsActivity : AppCompatActivity(), OnMapReadyCallback {
         }
     }
 
+    private val arrivalObserver = object : ArrivalController {
+        val arrivalOptions = ArrivalOptions.Builder()
+            .arriveInSeconds(60.0)
+            .build()
+        override fun arrivalOptions(): ArrivalOptions = arrivalOptions
+
+        override fun onArrival(routeLegProgress: RouteLegProgress): Boolean {
+            // This example shows you can use both time and distance.
+            // Move to the next step when the distance is small
+            return routeLegProgress.distanceRemaining() < 5.0
```

Comment:
Noticed that this keeps firing even after reaching the last leg arrival. Should we check for last leg downstream so arrivals are only fired once?

Reply:
it's a good question, the idea is once you return true it stops firing.

Reply:
Yeah, that makes sense but what about last leg? Should we return `true` internally?

Reply:
added an optional onRouteArrival function. a custom onStopArrival requires the developer handle moving to next leg

Reply:
or do decide when to return true from onStopArrival. there's not a action required onRouteArrival, so making a separate optional callback